cinema/views.py

```diff
@@ -21,24 +28,52 @@
 )
 
 
-class GenreViewSet(viewsets.ModelViewSet):
+class GenreViewSet(mixins.ListModelMixin,
+                   mixins.CreateModelMixin,
+                   viewsets.GenericViewSet):
     queryset = Genre.objects.all()
     serializer_class = GenreSerializer
 
+    def get_permissions(self):
+        if self.action == "create":
+            return [IsAdminUser()]
+        return [IsAuthenticated()]
+
 
-class ActorViewSet(viewsets.ModelViewSet):
+class ActorViewSet(mixins.ListModelMixin,
+                   mixins.CreateModelMixin,
+                   viewsets.GenericViewSet):
     queryset = Actor.objects.all()
     serializer_class = ActorSerializer
 
+    def get_permissions(self):
+        if self.action == "create":
+            return [IsAdminUser()]
+        return [IsAuthenticated()]
 
-class CinemaHallViewSet(viewsets.ModelViewSet):
+
+class CinemaHallViewSet(mixins.ListModelMixin,
+                        mixins.CreateModelMixin,
+                        viewsets.GenericViewSet):
     queryset = CinemaHall.objects.all()
     serializer_class = CinemaHallSerializer
 
+    def get_permissions(self):
+        if self.action == "create":
+            return [IsAdminUser()]
+        return [IsAuthenticated()]
```

Comment:
The `get_permissions` method in `CinemaHallViewSet` restricts the `create` action to admins only. The task specifies that both list and create actions should be available to all users. Please modify the permissions to meet this requirement.